test/document_stores/test_elasticsearch.py

```diff
@@ -348,6 +357,34 @@ def test_get_document_by_id_excluded_meta_data_has_no_influence(self, mocked_doc
     @pytest.mark.unit
     def test_write_documents_req_for_each_batch(self, mocked_document_store, documents):
         mocked_document_store.batch_size = 2
-        with patch("haystack.document_stores.elasticsearch.bulk") as mocked_bulk:
+        with patch("haystack.document_stores.elasticsearch7.bulk") as mocked_bulk:
             mocked_document_store.write_documents(documents)
             assert mocked_bulk.call_count == 5
+
+    @pytest.mark.unit
+    @pytest.mark.skipif(ES_VERSION[0] != 7, reason="Only run for ES7")
+    def test_import_from_haystack_document_stores_es7(self):
+        from haystack.document_stores import ElasticsearchDocumentStore
+
+        assert ElasticsearchDocumentStore.__module__ == "haystack.document_stores.elasticsearch7"
+
+    @pytest.mark.unit
+    def test_import_from_haystack_document_stores_es_client_not_installed(self):
+        original_import = builtins.__import__
+
+        def side_effect(name, *args):
+            if name == "elasticsearch":
+                raise ModuleNotFoundError("No module named 'elasticsearch'", name="elasticsearch")
+            return original_import(name, *args)
+
+        # Imports are cached in sys.modules. To test the import error, we need to remove the haystack from the cache
+        with patch.dict(sys.modules, {}):
+            modules_to_remove = [module for module in sys.modules if module.startswith("haystack")]
+            for module in modules_to_remove:
+                del sys.modules[module]
+            # mock the import statement to raise an error when trying to import elasticsearch
+            with patch("builtins.__import__", side_effect=side_effect):
+                with pytest.raises(ImportError, match="Run 'pip install farm-haystack\[elasticsearch\]'"):
+                    from haystack.document_stores import ElasticsearchDocumentStore
+
+                    doc_store = ElasticsearchDocumentStore()
```

Comment:
Are we testing `lazy_imports` here?

Reply:
We're testing that we reach [this line](https://github.com/deepset-ai/haystack/blob/9bc2f5e5f33fcffeea6344de38a7185919acb419/haystack/document_stores/__init__.py#L27) if the elasticsearch client is not reached and therefore import `elasticsearch7.ElasticsearchDocumentStore` and print the correct error message.